guides/Git and Code Reviews.md

```diff
@@ -0,0 +1,80 @@
+  refs/heads/ | sh
+# Set rebase=true for new branches in the future
+git config branch.autosetuprebase always
+```
+
+## Pull and rebase often
+
+Keep your master branch up to date. If you make your master branch use rebasing by default (see above), just run `git pull` on master.
+
+Rebasing your feature branches often (`git rebase master`) also increases the likelihood your code works with master and decreases the likelihood of a large merge conflict instead of smaller ones.
+
+If you’ve rebased a branch that you’ve sent for code review, you’ll need to force-push it to the remote branch on GitHub with `git push` `--``force`. Universe is configured to block accidental force-pushes to master.
+
+## Commit often
+
+On larger projects, send PRs and land them on master often. This makes each PR easier to review and gives you faster feedback on whether you’re working in the right direction. Frequent commits also reduce large spikes in the time you’re asking from your reviewer and reduce sudden changes to the codebase that may break something. They also help reduce the size of merge conflicts your teammates may encounter after pulling and rebasing on top of your changes. Keep your feature branches close to master.
+
+## Squash before pushing
+
+If you have more than one commit, squash them together before pushing if they are all part of the same idea. For example, commits that fix small bugs or address code review feedback usually should be squashed in the same commit. This makes it easier to read our commit log and revert commits. If some groups of your commits make more sense in isolation, then keep them separate.
+
+## Code behind feature flags
+
+Use simple flags — even a Boolean variable at the top of your file — to gate code that isn’t ready for production but that you still are committing to master. This way you can commit often without necessarily putting your code into action. It also lends itself to code that is easier to enable or disable experimentally without a lot of commitment.
+
+## Communicate with test plans
+
+Test plans are often effective at communicating the effects of your change. Write test plans as if you were explaining to a knowledgeable teammate how to test your change. Sometimes a test plan might just be “I ran the unit tests,” and other times they may be a manual QA process.
+
+Test plans communicate the scope of your change and what to look for when reviewing your code or modifying your code in the future. Someone who is working on the same code may look through the commit history to understand your code better, and test plans help them understand what to look out for when they make their own changes. Similarly, a test plan helps calibrate your code reviewer’s confidence.
+
+----------
+
+# Code Reviews
+
+We use GitHub pull requests to send code for review. Code reviews help establish a better shared understanding of the code between the author and the reviewer(s). We use code reviews more as a tool than a rule.
```

Comment:
Some of this section only applies internally - might be worth clarifying what should happen in the case of external PRs, too

Reply:
I added another guide in CONTRIBUTING.md that applies to external contributions. This will show up when people create a PR too.